### PR TITLE
qbs: Update to 2.0.2

### PR DESCRIPTION
**Pull request by ABBAPOH**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 13.3.1 22E772610a x86_64
Xcode 14.3 14E222b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by reneeotten — COMMENTED** [review]
see PR #16751 where a change to Qt6 was attempted. That should work now as I commented there, so perhaps incorporate that as well here?

**Comment by ABBAPOH** [comment]
Hi. Last time I tried Qt6 build, it failed because we kinda bootstrap qbs to build some stuff and it fails without setting DYLD_FRAMEWORK_PATH properly. Which should probably be set in upstream, but I was a bit lazy to fix it.
tl;dr: let's merge this and port to qt6 separately
